_rules/SC4-1-2-role-has-required-states-and-properties.md

```diff
@@ -0,0 +1,131 @@
+---
+name: Role has required states and properties
+description: |
+  Elements that has a semantic role must also specify all required states and properties
+  
+success_criterion:
+- 4.1.2 # Name, Role, Value
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Anne Thyme Nørregaard
+---
+
+## Test procedure
+
+### Applicability
+
+Any HTML or SVG element that is [exposed to assistive technologies](#exposed-to-assistive-technologies) and has an [explicit semantic role](#semantic-role), except if the element has an [implicit semantic role](#implicit-role) that is identical to the explicit semantic role. 
+
+### Expectation
+
+For each test target all attributes listed under [required states and properties](https://www.w3.org/TR/wai-aria/#requiredState) in [WAI-ARIA](https://www.w3.org/TR/wai-aria) for that role are present and [non-empty](#non-empty), unless they have a default value listed under [implicit value for role](https://www.w3.org/TR/wai-aria-1.1/#implictValueForRole).
+
+**Note:** This rule does not test whether the required states and properties has a correct value, only that the attributes are present and [non-empty](#non-empty).
+
+## Assumptions
```

Comment:
This section seems really long. Can it be cut down to the essential bits? For example, do we need the first two paragraphs? They seem to re-explain what is already defined by the applicability and expectation, or what is stated later in the accessibility support section. Putting this information in to different places will likely lead to confusion.

Reply:
I second that, lets see if we can simplify this.

Reply:
I removed almost all of it. Let me know, if we still needed any of the bits.

Reply:
This assumption is confusing to me for the following reasons:
1 - The applicability is for elements with an explicit semantic role. This assumption applies to elements with implicit semantic roles. 
2 - The assumption's reason to exist seems to be to justify the applicability. If that is case, then why isn't it just a note on the applicability?
3 - It is hard to read. There are too many negatives (no element that has an implicit semantic role + does not define native attributes). Can't this be rephrased?

Reply:
@carlosapaduarte,
1 - Assumption changed to make relation between rule and assumption clearer
2  - My impression is that all assumptions that could change the outcome of the rule if they were not true should be listed under Assumptions, whereas notes are more informative.
3 - Tried rephrasing. Hope the meaning is conserved.

Reply:
Awesome work cutting this down!